aion/lib/aion/room_channel/room.ex

```diff
@@ -73,15 +87,33 @@ defmodule Aion.RoomChannel.Room do
 
   @spec get_current_question(__MODULE__.t) :: Question.t
   def get_current_question(room) do
-    room.question
+    room.current_question
   end
 
-  @spec get_new_question_with_answers(integer) :: %{question: Question.t, answers: list Answer.t}
-  defp get_new_question_with_answers(room_id) do
-      question = Question.get_random_question(room_id)
-      answers = Answer.get_answers(question.id)
-      Logger.debug fn -> "Answers: #{inspect(Enum.map(answers, fn answer -> answer.content end))}" end
+  @spec get_new_question_with_answers((list Question.t), integer) :: new_question_set
+  defp get_new_question_with_answers(questions, room_id) do
+    case questions do
+      [last_question] ->
+        remaining_questions = fetch_questions(room_id)
+        get_new_question_set(last_question, remaining_questions)
+
+      [next_question | remaining_questions] ->
+        get_new_question_set(next_question, remaining_questions)
+
+      [] ->
+        %{questions: [], current_question: nil, answers: []}
+    end
+  end
+
+  @spec get_new_question_set(Question.t, list Question.t) :: new_question_set
+  defp get_new_question_set(current_question, remaining_questions) do
+    answers = Answer.get_answers(current_question.id)
+    Logger.debug fn -> "Answers: #{inspect(Enum.map(answers, fn answer -> answer.content end))}" end
+    %{questions: remaining_questions, current_question: current_question, answers: answers}
+  end
 
-      %{question: question, answers: answers}
+  @spec fetch_questions(integer) :: list Question.t
+  defp fetch_questions(room_id) do
```

Comment:
![image](https://user-images.githubusercontent.com/15965147/31360881-ba5ca9e4-ad50-11e7-8884-1737927be16b.png)

Reply:
I decided to leave it as it is

Reply:
I'm not sure if that's a good idea, since Elixir provides a way to deal with these cases. I've done some research to learn what's the best case here and here's what I've found:
[https://stackoverflow.com/a/40449310](Stack answer)

it could be then written pretty neatly like:
```
defdelegate fetch_questions, to: Question, as: :get_questions_by_room_id
```
What I find good about it is the fact that you don't need to specify the arguments anywhere, because you state clearly that the are in the *exact* same order and quantity. And this is exactly what you want to do here.

Reply:
Does it make fetch_questions private ?

Reply:
Let me paste some docs here:
![image](https://user-images.githubusercontent.com/15965147/31461353-938db0a4-aec9-11e7-85c0-ef06849588e3.png)

As it suggests, if you want to achieve a private function here, you should use `import`. You won't be able to change the function's name, though. 

If you care about the naming and you really want to have something else than `get_questions_by_room_id`, you could always consider naming the original function in a more versatile way :smiley: 

Another way is to define a lambda that would work as an alias only.